tests/unit/test_util.py

```diff
@@ -266,6 +268,53 @@ def _mock(x=1.0, *, a=2.0):
         # `post_time` is upper bounded by `pre_time - wait_time`.
         assert post_time < (pre_time - wait_time)
```

Comment:
Add an assertion that `post_time > 0`. I raise this because the post times I receive when testing `speed_comparison_test` in the console are all zero. A quick scan of the docstring for `time.time` suggests you may not be able to rely on it for fractions of a second, so you may need to find an alternative timing function here.